### DISCUSSION TITLE
Pie chart clusters

### DISCUSSION THREAD

**Discussion by Mstolarek** [rnmapbox/maps#2727] [Q&A]
Hello everyone

I am trying to recreate this behavior from mapbox GL JS docs 

https://docs.mapbox.com/mapbox-gl-js/example/cluster-html/?q=querySourceFeatures&size=n_10_n

I want to show 3 pie fragments : [available, unavailable, occupied]

i store my available , unavailalbe and occupied markers count as clusterProperties in <MapboxGL.ShapeSource />

Assuming that i am able to dynamically create properly filled pie chart in < View /> component and forward them as cluster icon through <MapboxGL.Image /> , is there any way i can get my markers count from clusterProperties to my < View /> component ?

using shapeSource.features() reffering to my <MapboxGL.ShapeSource /> seams like a good idea , but shapeSource.features() just stop execution and doesnt eaven give error in return :(

Please help <3

**Comment by mfazekas** [comment]
@Mstolarek pls report an issue about shapeSource.features(). Also you could create an iconImage: with expression that includes available, etc numbers. Then use Images's onImageMissing to generate the needed images on demand.
